### ISSUE TITLE
Push notifications not functioning (SYN-679)

### ISSUE THREAD

**Issue by matrixbot** [matrix-org/synapse#1363]
I have had a number of pushers entries simply fail to work, since a few weeks now, at least.

synapse=# select * from pushers;
 480 | @&#8203;rrix:whatthefuck.computer |         6994 | mobile_1245475184 | http | im.vector.app.android      | im.vector.alpha | SM-T710             | {REDACTED} | 1460398004333 | en   | {"url":"https://matrix.org/_matrix/push/v1/notify"} | s2199039_61590480_217177_207_0 | 1460419050527 |              
 470 | @&#8203;rrix:whatthefuck.computer |         7430 | mobile_1245475184 | http | org.matrix.console.android | org.matrix.androidsdk.alpha | SM-T710             | {REDACTED} | 1460095151818 | en   | {"url":"https://matrix.org/_matrix/push/v1/notify"} | s2199039_3770_217177_209_5     | 1460419050515 |              
 118 | @&#8203;rrix:whatthefuck.computer |          926 | mobile_1245475184 | http | org.matrix.console.android | org.matrix.androidsdk.alpha | LG-V498             | {REDACTED} | 1455593485374 | en   | {"url":"https://matrix.org/_matrix/push/v1/notify"} | s2199039_366925_217177_171_0   | 1460419050530 |              
(3 rows)

And then I deleted those and created two new ones which also fail:

 470 | @&#8203;rrix:whatthefuck.computer |         6994 | mobile_1245475184 | http | im.vector.app.android      | im.vector.alpha             | SM-T710             | {REDACTED | 1460420382598 | en   | {"url":"https://matrix.org/_matrix/push/v1/notify"} | s2206303_64684_7_219498_209_5 | 1460490147070 |              
 471 | @&#8203;rrix:whatthefuck.computer |         7430 | mobile_1245475184 | http | org.matrix.console.android | org.matrix.androidsdk.alpha | SM-T710             | {REDACTED} | 1460420443804 | en   | {"url":"https://matrix.org/_matrix/push/v1/notify"} | s2206303_64709_8_219498_209_5 | 1460490147059 |              
(2 rows)

Let me know if you need the REDACTED bits or anything else to help debug; Matthew was pretty sure bad things could be done with them, so I'm not putting them in Jira. 
This is also happening with the same device on the matrix.org HS, so maybe it's something tied to my device specifically? (bad GCM token or something?)

(Imported from https://matrix.org/jira/browse/SYN-679)

(Reported by Ryan Rix)

**Comment by paulnivin** [comment]
I believe I have the same issue. Let me know if you'd like for me to open a new issue.

I'm running a Synapse homeserver (0.18.5) and I'm unable to receive push notifications on Riot for Android (0.6.5 / Google Play) if Riot is not running. With "Enable background sync" enabled, it's more likely that notifications will arrive, but with it disabled, if Riot is not running, I don't receive notifications until Riot is launched. Force stop after enabling "Enable background sync" results in the same broken behavior.

This is on a stock Pixel XL running Android 7.1.1. "Notification Targets" shows "Pixel XL / im.vector.alpha".

Server logs show that Synapse is successfully calling matrix.org for the push notifications:
```
synapse.http.client: [] Sending request POST https://matrix.org/_matrix/push/v1/notify
synapse.http.client: [] Received response to  POST https://matrix.org/_matrix/push/v1/notify: 200
```

It appears that GCM is not waking Riot (assuming that GCM events are even being received for Riot).

Let me know how I can help further debug.

**Comment by brycied00d** [comment]
For what it's worth, I ran into a similar issue -- push notifications from matrix.org to my Nexus 5/Riot 0.6.5 (GPlay) had worked, but after signing-out and then into my new homeserver push notifications no longer worked. I confirmed that my homeserver was pushing notifications to matrix.org, and that my phone was indeed listed as a notification target.  
For me, the solution was to sign-out and "Clear Data," then sign back in. I guess that Riot didn't deregister/reregister with GCM with simply signing out and back in. (I never did bother going so far as to logcat)
